.github/workflows/build_validation_workflow.yml

```diff
@@ -2,18 +2,76 @@ name: Build Validation Workflow
 
 on:
   workflow_call:
+    inputs:
+      image_name:
+        type: string
+        description: "Execution Environment"
+        required: false
+      model_type:
+        type: string
+        description: "type of model to execute"
+        required: false
+      IS_DOCKER:
+        type: boolean
+        description: "Is Docker used for build validation?"
+        required: false
+        default: false
 
 env:
   PYTHON_VERSION: "3.9"
 
 jobs:
+  run-unit-tests-docker:
+    if: ${{ inputs.IS_DOCKER == true }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v1
+
+      - uses: actions/setup-python@v5
+        with:
+          python-version: ${{ env.PYTHON_VERSION }}
+
+      - name: Load Python Dependencies
+        shell: bash
+        run: |
+          set -e
+          python -m pip install --upgrade pip
+          python -m pip install --upgrade -r .github/requirements/build_validation_requirements.txt      
+      - name: Lint Code
+        run: flake8
+
+      - name: Build Docker container
```

Comment:
similar comment to capture the need to explore how to cache this as a future GH issue / story